File: python/euler/bin/p0045.py
# project euler: problem 45

# H_{n} = T_{2n-1}

from itertools import count

from euler.lib.util import is_pentagonal


def compute() -> str:
    for n in count(144):
        hex_num = n * (2 * n - 1)
        if is_pentagonal(hex_num):
            return str(hex_num)

    assert False, 'unreachable!'


def solve() -> str:
    return compute()

File: typescript/euler/bin/p0045.ts
// project euler: problem 45

/*
  H_{n} = T_{2n-1}
*/

import { isPentagonal } from "../lib/math.ts";

export const compute = (): string => {
  const hexNum = (n: number): number => n * (2 * n - 1);

  let d = 144;
  while (true) {
    const n = hexNum(d);
    if (isPentagonal(n) === true) {
      return String(n);
    }
    d += 1;
  }
};

export const solve = (): string => compute();
